sk-csharp-chatgpt-plugin/semantic-functions-generator/SemanticSkillGenerator.cs

```diff
@@ -0,0 +1,177 @@
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using AIPlugins.AzureFunctions.Extensions;
+
+namespace {rootNamespace};
+
+public class {folderName}
+{{
+    private readonly ILogger _logger;
+    private readonly IAIPluginRunner _pluginRunner;
+
+    public {folderName}(IAIPluginRunner pluginRunner, ILoggerFactory loggerFactory)
+    {{
+        this._pluginRunner = pluginRunner;
+        this._logger = loggerFactory.CreateLogger<{folderName}>();
+    }}
+
+    {functionsCode}
+}}";
+    }
+
+    private static string? GenerateFunctionSource(AdditionalText promptFile, AdditionalText configFile)
+    {
+        // Get the function name from the directory name
+        string? functionName = Path.GetFileName(Path.GetDirectoryName(promptFile.Path));
+        if (string.IsNullOrWhiteSpace(functionName)) { return null; }
+
+        string? metadataJson = configFile.GetText()?.ToString();
+        if (string.IsNullOrWhiteSpace(metadataJson)) { return null; }
+
+        // Get the function description from the config file
+        PromptConfig? config = JsonConvert.DeserializeObject<PromptConfig>(metadataJson!);
+        if (config == null) { return null; }
+
+        string descriptionProperty = string.IsNullOrWhiteSpace(config.Description)
+            ? string.Empty
+            : $@", Description = ""{config.Description}""";
+
+        string parameterAttributes = GenerateParameterAttributesSource(config.Input?.Parameters);
+
+        return $@"
+    [OpenApiOperation(operationId: ""{functionName}"", tags: new[] {{ ""{functionName}"" }}{descriptionProperty})]{parameterAttributes}
+    [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: ""text/plain"", bodyType: typeof(string), Description = ""The OK response"")]
+    [Function(""{functionName}"")]
+    public Task<HttpResponseData> {functionName}([HttpTrigger(AuthorizationLevel.Anonymous, ""post"")] HttpRequestData req)
+    {{
+        this._logger.LogInformation(""HTTP trigger processed a request for function {functionName}."");
+        return this._pluginRunner.RunAIPluginOperationAsync(req, ""{functionName}"");
+    }}";
+    }
+
+    private static string GenerateParameterAttributesSource(
+        List<PromptConfig.ParameterConfig>? parameters)
+    {
+        string inputDescription = string.Empty;
+        StringBuilder parameterStringBuilder = new();
+
+        if (parameters != null)
+        {
+            foreach (var parameter in parameters)
+            {
+                if (parameter.Name.Equals("input", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    // "input" is a special parameter that is handled differently. It must be added as
+                    // the body attribute.
```

Comment:
Somewhere else we are allowing the `input` to be passed as a request parameter, so is this comment incorrect?